std/hash/poseidon_bn254.asm

```diff
@@ -1,18 +1,18 @@
 use std::array;
 use std::utils::unchanged_until;
 
-// Implements the Poseidon permutation for the BN254 curve.
+// Implements the poseidon permutation for the BN254 curve.
 // Note that this relies on the trace table being non-wrapping, so it will
 // only work with the Halo2 backend (which is the only backend that supports
 // the BN254 curve).
-machine PoseidonBN254(FIRSTBLOCK, operation_id) {
+machine PoseidonBN254(LASTBLOCK, operation_id) {
```

Comment:
Actually undoing the change that the latch is on the first row, just for this example.

The problem is that with the latch on the first row, the input is only available on the first row, so we'd have to read the entire input from memory in one time step.